src/XR/native/nativeXRFrame.ts

```diff
@@ -0,0 +1,92 @@
+import { RegisterNativeTypeAsync } from "../../Engines/nativeEngine";
+
+/** @hidden */
+interface INativeXRFrame extends XRFrame {
+    // Native-only helper functions
+    getPoseData: (space: XRSpace, baseSpace: XRReferenceSpace, vectorBuffer: ArrayBuffer, matrixBuffer: ArrayBuffer) => XRPose;
+}
+
+/** @hidden */
+export class NativeXRFrame implements XRFrame {
+    private readonly _xrTransform = new XRRigidTransform();
+    private readonly _xrPose: XRPose = {
+        transform: this._xrTransform,
+        emulatedPosition: false
+    };
+    // Enough space for position, orientation
+    private readonly _xrPoseVectorData = new Float32Array(4 * (4 + 4));

```

Comment:
this should just be 8 right? this is the number of elements and not byteLength?

```suggestion
    private readonly _xrPoseVectorData = new Float32Array(4 + 4);
```